k3k-kubelet/translate/host.go

```diff
@@ -96,10 +99,13 @@ func (t *ToHostTranslater) TranslateName(namespace string, name string) string {
 	// - a valid k8s name
 	// - idempotently calculatable
 	// - unique for this combination of name/namespace/cluster
-	namePrefix := fmt.Sprintf("%s-%s-%s", name, namespace, t.ClusterName)
+	namePrefix := name + "-" + namespace + "-" + t.ClusterName
+
 	// use + as a separator since it can't be in an object name
-	nameKey := fmt.Sprintf("%s+%s+%s", name, namespace, t.ClusterName)
+	nameKey := name + "+" + namespace + "+" + t.ClusterName
+
```

Comment:
In this case (with 2+ args) I find the `Sprintf` more readable, but that's a personal preference.

Reply:
The trade off is an insane amount of heap and garbage collector pressure for no real gain when just concatenating strings.

Reply:
That's right. I'm still in favor of a better readability over small optimizations but in this case we are not "obfuscating" too much. :)

FYI (and future-me): I didn't know about the `-gcflags "-m -l"` flags. I did some experiment and if you create a small main.go and you run it with `go run -gcflags "-m -l" main.go` it will tell you if the variable escapes to the heap or not! Pretty cool.